lib/playground.dart

```diff
@@ -36,124 +46,132 @@ import 'sharing/gist_storage.dart';
+  DElement editorPanelHeader;
+  DElement editorPanelFooter;
+  MDCMenu samplesMenu;
+  MDCMenu moreMenu;
+  Dialog dialog;
+  NewPadDialog newPadDialog;
+  DElement titleElement;
+  MaterialTabController webLayoutTabController;
+  DElement webTabBar;
+  DElement webOutputLabel;
+
+  Splitter splitter;
+  Splitter rightSplitter;
+  bool rightSplitterConfigured = false;
+  TabExpandController tabExpandController;
+  AnalysisResultsController analysisResultsController;
 
-  DButton runButton;
-  DButton formatButton;
-  DOverlay overlay;
   DBusyLight busyLight;
   DBusyLight consoleBusyLight;
+
   Editor editor;
   PlaygroundContext _context;
   Future _analysisRequest;
-  MutableGist editableGist = MutableGist(Gist());
-  final _gistStorage = GistStorage();
-  DContentEditable titleEditable;
-
-  TabController sourceTabController;
-  TabController outputTabController;
-  SharingDialog sharingDialog;
-  KeysDialog settings;
+  Layout _layout;
 
-  // We store the last returned shared gist; it's used to update the url.
+  // The last returned shared gist used to update the url.
   Gist _overrideNextRouteGist;
   DocHandler docHandler;
 
-  Playground() {
-    sourceTabController = TabController();
-    for (String name in ['dart', 'html', 'css']) {
-      sourceTabController.registerTab(
-          TabElement(querySelector('#${name}tab'), name: name, onSelect: () {
-        Element issuesElement = querySelector('#issues');
-        issuesElement.style.display = name == 'dart' ? 'block' : 'none';
-        ga.sendEvent('edit', name);
-        _context.switchTo(name);
-      }));
-    }
-
-    overlay = DOverlay(querySelector('#frame-overlay'));
-
-    sharingDialog = SharingDialog(this, this);
-
-    DButton newButton = DButton(querySelector('#newbutton'));
-    OkCancelDialog newDialog = OkCancelDialog(
-        'Create New Pad', 'Discard changes to the current pad?', createNewGist,
-        okText: 'Discard');
-    newButton.onClick.listen((_) {
-      newDialog.show();
-    });
-
-    DButton resetButton = DButton(querySelector('#resetbutton'));
-    OkCancelDialog resetDialog = OkCancelDialog(
-        'Reset Pad', 'Discard changes to the current pad?', _resetGists,
-        okText: 'Discard', cancelText: 'Cancel');
-    resetButton.onClick.listen((_) {
-      resetDialog.show();
-    });
+  Console _leftConsole;
+  Console _rightConsole;
+  Counter unreadConsoleCounter;
 
-    editableGist.onDirtyChanged.listen((val) {
-      resetButton.disabled = !val;
+  Playground() {
+    _initModules().then((_) {
+      _initPlayground();
+      _initDialogs();
+      _initBusyLights();
+      _initGistNameHeader();
+      _initGistStorage();
+      _initLayoutDetection();
+      _initButtons();
+      _initLabels();
+      _initSamplesMenu();
+      _initMoreMenu();
+      _initSplitters();
+      _initTabs();
+      _initLayout();
+      _initConsoles();
     });
+  }
 
-    DButton shareButton = DButton(querySelector('#sharebutton'));
```

Comment:
Ooh, I can probably rip out some of the old sharing stuff from the gist-related classes now. This was the only place that referenced it, I think.